vignettes/functionality-flow.Rmd

```diff
@@ -0,0 +1,185 @@
+    (Laboratoriedatabasens Forskertabel)
+-   The Danish National Patient Register, *DNPR*
+    (Landspatientregisteret, *LPR*)
+-   The Register of Pharmaceutical Sales (Lægemiddelstatistikregisteret)
+-   The Danish Civil Registration System (CPR-registerets
+    befolkningstabel, *BEF*)
+-   The National Health Insurance Service Register
+    (Sygesikringsregisteret)
+
+This component will have one user-facing function:
+`extract_diabetes_population`. All five data sources are used as input
+for this function.
+
+As described in the [design](design.Rmd) vignette, the inclusion date is
+the date of the second inclusion criteria. Inclusion events include:
+
+-   HbA1c measurements of ≥48 mmol/mol
+-   Hospital diagnoses of diabetes
+-   Diabetes-specific services received at podiatrists
+-   Purchase of glucose-lowering drugs (GLD)
+
+Note that HbA1c tests and GLD purchases during pregnancies are excluded.
+See section [Exclusion: HbA1c tests and GLD purchases during estimated
+pregnancy windows](#exclusion-pregnancy-windows) for how pregnancy
+windows are estimated. Glucose-lowering brand drugs for weight loss and
+metformin purchases for women below age 40 are excluded as well.
+
+This results in the functionality flow for extracting the diabetes
+population seen below. All functions take a `data.frame` as input and
+outputs a `data.frame`.
+
+![Flow of functions for extracting a diabetes population using the
+`osdc` package. The dark grey box denotes the user-facing function,
+while the remaining boxes are internal functions. Green and red boxes
+represent filtering functions (inclusion and exclusion events,
+respectively).](images/diabetes-population-functions.png)
+
+### Inclusion event: HbA1c tests above 48 mmol/mol
+
+The function `include_hba1c_tests_over_48_mmol_per_mol` uses RLRR as
+data source to extract all events of tests above 48 mmol/mol.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### Inclusion event: Hospital diagnosis of diabetes
+
+The function `include_diabetes_diagnoses` uses the hospital contacts
+from DNPR to include all dates of diabetes diagnoses. Diabetes diagnoses
+from both ICD 8 and ICD 10 are included.
+
+This function contains two helper functions:
+
+-   `extract_diabetes_diagnoses_icd_10`
+-   `extract_diabetes_diagnoses_icd_8`
+
+<!-- TODO: Add details on how this filtering should be done, e.g., diagnosis codes -->
+
+<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+
+### Inclusion event: Diabetes-specific podiatrist services
+
+The function `include_diabetes_specific_podiatrist_services` uses the
+Health Service Register to extract the dates of all diabetes-specific
+podiatrist services.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### Inclusion event: GLD purchases
+
+The function `include_gld_purchases` uses Lægemiddelsdatabasen to
+extract the dates of all GLD purchases.
+
+<!-- TODO: Add English translation of data source -->
+
```

Comment:
This can be easily done with the `variable_description` dataframe. Which I will add shortly.